Dockerfile.linux-build

```
@@ -0,0 +1,83 @@
+# Dockerfile for cross-compiling canined for Linux with CGO/ledger support
+# Runs natively on ARM Mac, cross-compiles to x86_64
+FROM golang:1.23.1-bookworm
+
+# Build arguments
+ARG GOARCH=amd64
+ARG GOOS=linux
+ARG LEDGER_ENABLED=true
+
+# Install native and cross-compilation build dependencies
+RUN dpkg --add-architecture amd64 && \
```

Hadolint (2.14.0)

[warning] Pin versions in apt get install. Instead of `apt-get install <package>` use `apt-get install <package>=<version>`

(DL3008)